spark/spark-tensorflow-connector/src/test/scala/org/tensorflow/spark/datasources/tfrecords/PropagateSuite.scala

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2016 The TensorFlow Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.tensorflow.spark.datasources.tfrecords
+
+import java.nio.file.Files
+
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.catalyst.expressions.GenericRow
+import org.apache.spark.sql.types._
+
+class PropagateSuite extends SharedSparkSessionSuite {
+
+  val testRows: Array[Row] = Array(
+    new GenericRow(Array[Any](11, 1, 23L, 10.0F, 14.0, List(1.0, 3.0), "r1")),
+    new GenericRow(Array[Any](21, 2, 24L, 12.0F, 15.0, List(2.0, 3.0), "r2")),
+    new GenericRow(Array[Any](31, 3, 25L, 14.0F, 16.0, List(3.0, 3.0), "r3")))
+  val schema = StructType(List(StructField("id", IntegerType),
+    StructField("IntegerTypeLabel", IntegerType),
+    StructField("LongTypeLabel", LongType),
+    StructField("FloatTypeLabel", FloatType),
+    StructField("DoubleTypeLabel", DoubleType),
+    StructField("VectorLabel", ArrayType(DoubleType, true)),
+    StructField("name", StringType)))
+
+
+  "Propagate" should {
+    "write data locally" in {
+      // Create a dataframe with 2 partitions
+      val rdd = spark.sparkContext.parallelize(testRows, numSlices = 2)
+      val df = spark.createDataFrame(rdd, schema)
+
+      // Write the partitions onto the local hard drive. Since it is going to be the
+      // local file system, the partitions will be written in the same directory of the
+      // same machine.
+      // In a distributed setting though, two different machines would each hold a single
+      // partition.
+      val localPath = Files.createTempDirectory("spark-connector-propagate").toAbsolutePath.toString
+      Propagate.writePartitionsLocal(df, localPath)
+
```

Comment:
Add localWrites as an boolean option so that users can access it using a consistent API, e.g., videoDf.write.format("tfrecords").option("recordType", "Example").option("localWrites", "true").save(localPath) where the default is false. You can change the name of option to whatever makes the most sense (I'm not great at picking names). You would need to refactor the code in DefaultSource.createRelation() to support the local writes and the regular writes.

Reply:
Also add documentation for the option in the README

Reply:
I agree this is much better than having an extra function call.